src/core/lib/slice/slice_intern.cc

```diff
@@ -69,62 +64,27 @@ typedef struct {
 static static_metadata_hash_ent
     static_metadata_hash[4 * GRPC_STATIC_MDSTR_COUNT];
 static uint32_t max_static_metadata_hash_probe;
-static uint32_t static_metadata_hash_values[GRPC_STATIC_MDSTR_COUNT];
+uint32_t static_metadata_hash_values[GRPC_STATIC_MDSTR_COUNT];
 
-static void interned_slice_ref(void* p) {
-  interned_slice_refcount* s = static_cast<interned_slice_refcount*>(p);
-  GPR_ASSERT(gpr_atm_no_barrier_fetch_add(&s->refcnt, 1) > 0);
-}
+namespace grpc_core {
 
-static void interned_slice_destroy(interned_slice_refcount* s) {
-  slice_shard* shard = &g_shards[SHARD_IDX(s->hash)];
-  gpr_mu_lock(&shard->mu);
-  GPR_ASSERT(0 == gpr_atm_no_barrier_load(&s->refcnt));
+void InternedSliceRefcount::DestroyInstance() {
+  slice_shard* shard = &g_shards[SHARD_IDX(this->hash)];
+  MutexLock lock(&shard->mu);
+  // TODO(arjunroy) Really need this method
```

Comment:
??? Is this needed or deletable?

Reply:
@soheilhy thoughts on this?

Basically there's an assert here on destroy, which in the gpr_ref world we could manually check. But it's a private variable now.

Reply:
So for now I've commented out the check; to get it to work, we'd need to add a get() method to the grpc_core::RefCount.

Reply:
I would say this check is unnecessary, because it's an extra load, which is simply there for debugging. To elaborate, if we want to check that refcount is 0 when we are in destroy we would have to do that everywhere in gRPC. So, I would just drop it.

But I understand the need for get. We can add it to the ref count for debug builds.

Reply:
Ack; will just drop it.